src/storage/FileLoader.cpp

```diff
@@ -378,17 +413,26 @@ std::vector<uint8_t> FileLoader::loadSharedObjectWasm(const std::string& path)
 
 std::string FileLoader::getSharedObjectObjectFile(const std::string& realPath)
 {
-    boost::filesystem::directory_entry f(realPath);
+    std::filesystem::directory_entry f(realPath);
     const std::string directory = f.path().parent_path().string();
     const std::string fileName = f.path().filename().string();
 
     // Work out the final destination for the object file. This will be the
     // object path with the directory of the original file appended
-    boost::filesystem::path objPath(conf.objectFileDir);
-    objPath.append(directory);
+    std::filesystem::path objPath(conf.objectFileDir);
+    objPath += directory;
```

Comment:
`path.append` works slightly different in `std::filesystem`. Namely if appending two paths both of which start with a backslash, then the first one will be ignored, and the second one taken by an absolute path (unless otherwise stated).

See [this post for more details](https://stackoverflow.com/questions/48347642/differences-between-c14-stdexperimentalfilesystemv1-and-c17-stdfiles/48349966#48349966).

Reply:
OK sure, this seems like odd behaviour to me (and clearly to the `boost::filesystem` authors too), so could you put a comment above this as a warning? I also think you mean forward slash rather than backslash (unless you're running on windows 😄)